### PR TITLE
[#1279] feat(jdbc): MySQL supports auto_increment

### PR DESCRIPTION
**Pull request by Clearvive**
### What changes were proposed in this pull request?
1.Delete unnecessary code sections about the properties of the jdbc column
2.MySQL supports load properties of auto_increment

### Why are the changes needed?
Fix: #1279 

### Does this PR introduce _any_ user-facing change?
NO

### How was this patch tested?
UT

### CODE REVIEW

File: api/src/main/java/com/datastrato/gravitino/rel/types/Types.java [hunk 1]
```diff
@@ -923,4 +923,12 @@ public int hashCode() {
...
     }
   }
+
+  /**
+   * @param dataType The data type to check.
+   * @return True if the given data type is allowed to be an auto-increment column.
+   */
+  public static boolean isAllowAutoIncrement(Type dataType) {
+    return dataType instanceof IntegerType || dataType instanceof LongType;
+  }
```
Comment (diqiu50) [inline api/src/main/java/com/datastrato/gravitino/rel/types/Types.java:933]: Using 'isAllowAutoIncrement' as a member method in Types.java might not be appropriate. It is recommended to provide a Util class like `Strings` to implement this.

**Comment by jerryshao** [comment]
@FANNG1 @mchades can you please help to review.

File: catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/operation/MysqlTableOperations.java [hunk 2]
```diff
@@ -565,11 +548,10 @@ private StringBuilder appendColumnDefinition(JdbcColumn column, StringBuilder sq
       sqlBuilder.append("DEFAULT '").append(column.getDefaultValue()).append("'").append(SPACE);
     }
 
-    // Add column properties if specified
-    if (CollectionUtils.isNotEmpty(column.getProperties())) {
-      for (String property : column.getProperties()) {
-        sqlBuilder.append(property).append(SPACE);
-      }
+    // Add column auto-increment
+    if (column.autoIncrement()) {
```
Comment (FANNG1) [inline catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/operation/MysqlTableOperations.java:552]: I prefer to support column.autoIncrement in Mysql, because it's straight forward

**Comment by FANNG1** [comment]
suggest to sperate into 2 PRs:
1. remove unnesseary code
2. auto_increment support for MySQL, including, create, load, alter operations and tests.

File: catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/operation/MysqlTableOperations.java [hunk 3]
```diff
@@ -565,11 +548,10 @@ private StringBuilder appendColumnDefinition(JdbcColumn column, StringBuilder sq
...
+      throw new UnsupportedOperationException(
+          "Mysql auto-increment columns need to be set as index columns");
```
Comment (mchades) [inline catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/operation/MysqlTableOperations.java:554]: The exception message looks quite strange.

Is this just a temporary exception? Will there be another PR to implement this?

**Comment by FANNG1** [comment]
LGTM

File: catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/operation/MysqlTableOperations.java [hunk 4]
```diff
@@ -132,6 +136,37 @@ protected String generateCreateTableSql(
     return result;
   }
 
+  private static void validateIncrementCol(JdbcColumn[] columns, Index[] indexes) {
```
Comment (FANNG1) [inline catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/operation/MysqlTableOperations.java:139]: please add some description

**Comment by jerryshao** [comment]
@FANNG1 you can go ahead if you feel OK.

**Review by FANNG1 — APPROVED** [review]
